include/mender-deployment-data.h

```diff
@@ -121,12 +132,17 @@ mender_err_t __mender_deployment_data_get_string(const mender_deployment_data_t
 #define mender_deployment_data_get_id(deployment_data, id) __mender_deployment_data_get_string(deployment_data, MENDER_DEPLOYMENT_DATA_KEY_ID, id)
 
 /**
- * @brief Get state name
+ * @warning Do not use this function directly.
+ */
+mender_err_t __mender_deployment_data_get_state(mender_deployment_data_t *deployment_data, const char *key, mender_update_state_t *state);
+
+/**
+ * @brief Get state
  * @param deployment_data Deployment data
- * @param name State name
+ * @param State state
  * @return MENDER_OK on success, otherwise MENDER_FAIL
  */
-#define mender_deployment_data_get_state(deployment_data, name) __mender_deployment_data_get_string(deployment_data, MENDER_DEPLOYMENT_DATA_KEY_STATE, state)
+#define mender_deployment_data_get_state(deployment_data, state) __mender_deployment_data_get_state(deployment_data, MENDER_DEPLOYMENT_DATA_KEY_STATE, state);
```

Comment:
Since this is a "function" with it's own definition, we don't really need to use this macro. And we know that the key is always `MENDER_DEPLOYMENT_DATA_KEY_STATE`, so we don't need to pass it as an argument.